websites/F/Free Web Novel/presence.ts

```diff
@@ -0,0 +1,185 @@
+
+	switch (true) {
+		case privacy: {
+			presenceData.largeImageKey = "incognito";
+
+			if (
+				(pathnames.includes(pathname) ||
+					document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) &&
+				showReading
+			)
+				presenceData.details = "Browsing...";
+			else if (
+				document.querySelector<HTMLAnchorElement>(
+					'[title="Read Next chapter"]'
+				) &&
+				showReading
+			) {
+				presenceData.details = "Reading...";
+				presenceData.smallImageKey = "open";
+			}
+			break;
+		}
+		case showLogo: {
+			presenceData.largeImageKey = "fwn_1024";
+			delete presenceData.smallImageKey;
+			delete presenceData.details;
+			delete presenceData.state;
+			delete presenceData.buttons;
+			break;
+		}
+		case pathname === "/": {
+			presenceData.details = "Browsing the homepage";
+			break;
+		}
+		case pathname.includes("/history/"): {
+			presenceData.details = "Browsing recently viewed novels";
+			break;
+		}
+		case pathname.includes("/latest-release-novel/"): {
+			presenceData.details = "Browsing latest released novels";
+			break;
+		}
+		case pathname.includes("/latest-novel/"): {
+			presenceData.details = "Browsing the latest novels";
+			break;
+		}
+		case pathname.includes("/completed-novel/"): {
+			presenceData.details = "Browsing fully finished novels";
+			break;
+		}
+		case pathname.includes("/most-popular-novel/"): {
+			presenceData.details = "Browsing the most popular novels";
+			break;
+		}
+		case pathname.includes("/genre/"): {
+			presenceData.details = `Checking out ${document
+				.querySelector<HTMLHeadingElement>('[class="tit"]')
+				.textContent.toLowerCase()}`;
+			break;
+		}
+		case pathname.includes("/author/"): {
+			presenceData.details = "Checking out an author";
+			presenceData.state = document.querySelector<HTMLAnchorElement>(
+				"body > div.header > div.cur > div > a:nth-child(5)"
+			).title;
+			if (showButtons) {
+				presenceData.buttons = [
+					{
+						label: "View Author",
+						url: location.href
+					}
+				];
+			}
+			break;
+		}
+		case pathname === "/search/": {
+			presenceData.details = "Searching for a novel";
+			break;
+		}
+		case pathname === "/contact.html": {
+			presenceData.details = "Viewing the contact page";
+			break;
+		}
+		case pathname === "/sitemap.xml": {
+			presenceData.details = "Viewing the site's sitemap";
+			break;
+		}
+		case pathname === "/privacy-terms-of-use.html": {
+			presenceData.details = "Reviewing the terms of use";
+			break;
+		}
+		default: {
+			if (document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) {
+				presenceData.largeImageKey = showCover
+					? `${
+							document.querySelector<HTMLImageElement>(
+								"body > div.main > div > div > div.col-content > div.m-info > div.m-book1 > div.m-imgtxt > div.pic > img"
+							).src
+					  }`
+					: "nocover";
+				presenceData.details = showReading ? "Viewing a novel" : null;
+				presenceData.state = showBook ? document.title.split("-")[0] : null;
```

Comment:
these should be ifs instead, no need to set these to null

Reply:
> these should be ifs instead, no need to set these to null

is it ok to leave these for the definition of `presenceData` or should those be changed as well?

![image](https://user-images.githubusercontent.com/96930330/168452987-dd1606ea-3c06-493f-9284-cf292e72e178.png)

Reply:
IMO you should set all the data and have a single if at the end to delete any unwanted data